### PR TITLE
Update links for https transition

### PR DESCRIPTION
**Pull request by rchild-okta**
## Description:

**NOTE: This will not be merged until we've cut over to https.**

- Updates config to https and removes hardcoded domains
- Fixes broken links found from htmlproofer. By default, it does not run on external domains (which was the case for all links that were previously hardcoded to http://developer.okta.com/foo), so these popped up when I switched to relative urls.

## Type of Pull Request:
- [x] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-127662](https://oktainc.atlassian.net/browse/OKTA-127662)

## How Has This Been Tested?
- [X] I have built this locally and verified that it does not break existing functionality.

## Primary Reviewers:
@lboyette-okta, @mauriciocastillosilva-okta

### CODE REVIEW

File: _source/_docs/api/getting_started/api_test_client.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ To use these collections, you'll need to set up your environment, import the col
 
 1. [Create an API token](getting_a_token.html){:target="_blank"}.
 2. [Install the Postman App](https://www.getpostman.com/apps){:target="_blank"}.
-3. Launch Postman and click the 'import button'. Select import from link, and paste this link into the textbox: `http://developer.okta.com/docs/api/postman/example.oktapreview.com.environment`
+3. Launch Postman and click the 'import button'. Select import from link, and paste this link into the textbox: `/docs/api/postman/example.oktapreview.com.environment`
```
Comment (lboyette-okta) [inline _source/_docs/api/getting_started/api_test_client.md]: We need to add this one back

File: _source/_docs/platform-release-notes/platform-release-notes2016-51.md [hunk 2]
```diff
@@ -27,4 +27,4 @@ To verify the current release for an org, click the **Admin** button and check t
 ### Looking for Something Else?
 
 * [Platform Release Note Index for 2016](platform-release-notes2016-index.html)
-* For changes outside the Okta platform, see the [Release Notes Knowledge Hub](http://support.okta.com/help/articles/Knowledge_Article/Release-Notes-Knowledge-Hub).
\ No newline at end of file
+* For changes outside the Okta platform, see the [Release Notes Knowledge Hub](http://support.okta.com/help/articles/Knowledge_Article/Release-Notes-Knowledge-Hub).
```
Comment (lboyette-okta) [inline _source/_docs/platform-release-notes/platform-release-notes2016-51.md]: Boy Scout nit: http://support.okta.com => https://support.okta.com

File: _source/_posts/2017-03-30-react-okta-sign-in-widget.md [hunk 3]
```diff
@@ -67,19 +67,19 @@ This little component doesn't *do* much but at least you now have a handle to ad
...
 import LoginPage from './components/auth/LoginPage';
-``` 
+```
 and then add the route inside the main route (the one with the path of "/")
 ```js
 <Route path="/login" component={LoginPage}/>
 ```
 ## Add the OpenID Connect Application in Okta
-In order to *use* Okta as your OpenID Connect provider for authentication, you’ll need to set up an application in the [Okta developer portal](http://developer.okta.com/).
+In order to *use* Okta as your OpenID Connect provider for authentication, you’ll need to set up an application in the [Okta developer portal](/).
```
Comment (lboyette-okta) [inline _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: `/` is `https://developer.okta.com` everywhere else

File: _source/_use_cases/api_security/index.html [hunk 4]
```diff
@@ -2,9 +2,9 @@
...
   <title>Redirecting…</title>
-  <link rel="canonical" href="http://developer.okta.com/use_cases/api_access_management/index.html">
-  <meta http-equiv="refresh" content="0; url=http://developer.okta.com/use_cases/api_access_management/index.html">
+  <link rel="canonical" href="/use_cases/api_access_management/index.html">
+  <meta http-equiv="refresh" content="0; url=/use_cases/api_access_management/index.html">
   <h1>Redirecting…</h1>
-  <a href="http://developer.okta.com/use_cases/api_access_management/index.html">Click here if you are not redirected.</a>
-  <script>location="http://developer.okta.com/use_cases/api_access_management/index.html"</script>
+  <a href="/use_cases/api_access_management/index.html">Click here if you are not redirected.</a>
+  <script>location="/use_cases/api_access_management/index.html"</script>
```
Comment (lboyette-okta) [inline _source/_use_cases/api_security/index.html]: Had to test this. I'd forgotten you can use relative paths in `location`

File: _source/documentation/stormpath-import.md [hunk 5]
```diff
@@ -279,7 +279,7 @@ What your imported Custom Data looks like depends primarily on whether is a simp
 <a name="simple-custom-data"></a>
 ##### Simple Custom Data
 
-Any Custom Data attributes are imported and added to a user's profile as custom schema properties. For more on the Okta User Profile Schema go here: http://developer.okta.com/docs/api/resources/schemas.html#schema-properties
+Any Custom Data attributes are imported and added to a user's profile as custom schema properties. For more on the Okta User Profile Schema go here: /docs/api/resources/schemas.html#schema-properties
```
Comment (lboyette-okta) [inline _source/documentation/stormpath-import.md]: Need the `https://developer.okta.com` back, or rephrase to use a markdown link instead.

File: _source/_docs/api/resources/resource-server-beta/required-config-changes-for-auth-server.md [hunk 6]
```diff
@@ -132,8 +132,8 @@ To change the well-known endpoint:
 1. Find the authorization server ID for your new authorization server. It's displayed in the **Metadata URI** field of the
 settings tab ( **Security > API > Authorization Server** ).
 
-2. Use the Discovery Document for [OpenID Connect clients](http://developer.okta.com/docs/api/resources/oidc.html#openid-connect-discovery-document)
-and [OAuth 2.0 clients](http://developer.okta.com/docs/api/resources/oauth2.md#authorization-server-metadata) to identify all endpoints that must be changed, as shown in the example below.
```
Comment (mauriciocastillosilva-okta) [inline _source/_docs/api/resources/resource-server-beta/required-config-changes-for-auth-server.md]: 👍 md -> html

File: _source/_use_cases/api_security/index.html [hunk 7]
```diff
@@ -2,9 +2,9 @@
 <html lang="en-US">
   <meta charset="utf-8">
   <title>Redirecting…</title>
-  <link rel="canonical" href="http://developer.okta.com/use_cases/api_access_management/index.html">
-  <meta http-equiv="refresh" content="0; url=http://developer.okta.com/use_cases/api_access_management/index.html">
+  <link rel="canonical" href="/use_cases/api_access_management/index.html">
+  <meta http-equiv="refresh" content="0; url=/use_cases/api_access_management/index.html">
```
Comment (mauriciocastillosilva-okta) [inline _source/_use_cases/api_security/index.html]: this are the only link and meta that use relative path. Should we add the whole url?

File: docs/api/resources/resource-server-beta/required-config-changes-for-auth-server.html [hunk 8]
```diff
@@ -364,8 +364,8 @@ <h3 id="step-five-change-your-clients-to-consume-the-new-endpoints">Step Five: C
 settings tab ( <strong>Security &gt; API &gt; Authorization Server</strong> ).</p>
   </li>
   <li>
-    <p>Use the Discovery Document for <a href="http://developer.okta.com/docs/api/resources/oidc.html#openid-connect-discovery-document">OpenID Connect clients</a>
-and <a href="http://developer.okta.com/docs/api/resources/oauth2.md#authorization-server-metadata">OAuth 2.0 clients</a> to identify all endpoints that must be changed, as shown in the example below.
```
Comment (mauriciocastillosilva-okta) [inline docs/api/resources/resource-server-beta/required-config-changes-for-auth-server.html]: 👍

**Pull request merged**